# import calc

from PyQt5 import QtCore, QtGui, QtWidgets

class Ui_MainWindow(object):
    def setupUi(self, MainWindow):
        MainWindow.setObjectName("MainWindow")
        MainWindow.resize(790, 464)
        MainWindow.setStyleSheet("background-color: rgb(46, 52, 54);")
        self.centralwidget = QtWidgets.QWidget(MainWindow)
        self.centralwidget.setObjectName("centralwidget")
        self.label = QtWidgets.QLabel(self.centralwidget)
        self.label.setGeometry(QtCore.QRect(10, 70, 771, 71))
        font = QtGui.QFont()
        font.setFamily("MathJax_Math")
        font.setPointSize(17)
        self.label.setFont(font)
        self.label.setStyleSheet("background-color: rgb(136, 138, 133);")
        self.label.setObjectName("label")
        self.zero = QtWidgets.QPushButton(self.centralwidget)
        self.zero.setGeometry(QtCore.QRect(350, 360, 101, 41))
        self.zero.setObjectName("zero")
        self.dub = QtWidgets.QPushButton(self.centralwidget)
        self.dub.setGeometry(QtCore.QRect(460, 360, 101, 41))
        self.dub.setObjectName("dub")
        self.result = QtWidgets.QPushButton(self.centralwidget)
        self.result.setGeometry(QtCore.QRect(570, 360, 101, 41))
        self.result.setStyleSheet("\n"
"color: rgb(120, 174, 213);")
        self.result.setObjectName("result")
        self.plus = QtWidgets.QPushButton(self.centralwidget)
        self.plus.setGeometry(QtCore.QRect(680, 360, 101, 41))
        self.plus.setStyleSheet("background-color: rgb(109, 123, 129);")
        self.plus.setObjectName("plus")
        self.button1 = QtWidgets.QPushButton(self.centralwidget)
        self.button1.setGeometry(QtCore.QRect(350, 310, 101, 41))
        self.button1.setObjectName("button1")
        self.button4 = QtWidgets.QPushButton(self.centralwidget)
        self.button4.setGeometry(QtCore.QRect(350, 260, 101, 41))
        self.button4.setObjectName("button4")
        self.button7 = QtWidgets.QPushButton(self.centralwidget)
        self.button7.setGeometry(QtCore.QRect(350, 210, 101, 41))
        self.button7.setObjectName("button7")
        self.button2 = QtWidgets.QPushButton(self.centralwidget)
        self.button2.setGeometry(QtCore.QRect(460, 310, 101, 41))
        self.button2.setObjectName("button2")
        self.button8 = QtWidgets.QPushButton(self.centralwidget)
        self.button8.setGeometry(QtCore.QRect(460, 210, 101, 41))
        self.button8.setObjectName("button8")
        self.button5 = QtWidgets.QPushButton(self.centralwidget)
        self.button5.setGeometry(QtCore.QRect(460, 260, 101, 41))
        self.button5.setObjectName("button5")
        self.button6 = QtWidgets.QPushButton(self.centralwidget)
        self.button6.setGeometry(QtCore.QRect(570, 260, 101, 41))
        self.button6.setObjectName("button6")
        self.button3 = QtWidgets.QPushButton(self.centralwidget)
        self.button3.setGeometry(QtCore.QRect(570, 310, 101, 41))
        self.button3.setStyleSheet("")
        self.button3.setObjectName("button3")
        self.button9 = QtWidgets.QPushButton(self.centralwidget)
        self.button9.setGeometry(QtCore.QRect(570, 210, 101, 41))
        self.button9.setObjectName("button9")
        self.minus = QtWidgets.QPushButton(self.centralwidget)
        self.minus.setGeometry(QtCore.QRect(680, 310, 101, 41))
        self.minus.setStyleSheet("background-color: rgb(109, 123, 129);")
        self.minus.setObjectName("minus")
        self.divivsion = QtWidgets.QPushButton(self.centralwidget)
        self.divivsion.setGeometry(QtCore.QRect(680, 210, 101, 41))
        self.divivsion.setStyleSheet("background-color: rgb(109, 123, 129);")
        self.divivsion.setObjectName("divivsion")
        self.miltiply = QtWidgets.QPushButton(self.centralwidget)
        self.miltiply.setGeometry(QtCore.QRect(680, 260, 101, 41))
        self.miltiply.setStyleSheet("background-color: rgb(109, 123, 129);")
        self.miltiply.setObjectName("miltiply")
        self.drop = QtWidgets.QPushButton(self.centralwidget)
        self.drop.setGeometry(QtCore.QRect(680, 160, 101, 41))
        self.drop.setStyleSheet("background-color: rgb(109, 123, 129);")
        self.drop.setObjectName("drop")
        self.leftBR = QtWidgets.QPushButton(self.centralwidget)
        self.leftBR.setGeometry(QtCore.QRect(460, 160, 101, 41))
        self.leftBR.setStyleSheet("background-color: rgb(109, 123, 129);")
        self.leftBR.setObjectName("leftBR")
        self.rightBR = QtWidgets.QPushButton(self.centralwidget)
        self.rightBR.setGeometry(QtCore.QRect(570, 160, 101, 41))
        self.rightBR.setStyleSheet("background-color: rgb(109, 123, 129);")
        self.rightBR.setObjectName("rightBR")
        self.ln = QtWidgets.QPushButton(self.centralwidget)
        self.ln.setGeometry(QtCore.QRect(240, 210, 101, 41))
        self.ln.setStyleSheet("background-color: rgb(109, 123, 129);")
        self.ln.setObjectName("ln")
        self.log = QtWidgets.QPushButton(self.centralwidget)
        self.log.setGeometry(QtCore.QRect(240, 260, 101, 41))
        self.log.setStyleSheet("background-color: rgb(109, 123, 129);")
        self.log.setObjectName("log")
        self.sqrt = QtWidgets.QPushButton(self.centralwidget)
        self.sqrt.setGeometry(QtCore.QRect(240, 310, 101, 41))
        self.sqrt.setStyleSheet("background-color: rgb(109, 123, 129);")
        self.sqrt.setObjectName("sqrt")
        self.square = QtWidgets.QPushButton(self.centralwidget)
        self.square.setGeometry(QtCore.QRect(240, 360, 101, 41))
        self.square.setStyleSheet("background-color: rgb(109, 123, 129);")
        self.square.setObjectName("square")
        self.sin = QtWidgets.QPushButton(self.centralwidget)
        self.sin.setGeometry(QtCore.QRect(130, 260, 101, 41))
        self.sin.setStyleSheet("background-color: rgb(109, 123, 129);")
        self.sin.setObjectName("sin")
        self.tg = QtWidgets.QPushButton(self.centralwidget)
        self.tg.setGeometry(QtCore.QRect(130, 360, 101, 41))
        self.tg.setStyleSheet("background-color: rgb(109, 123, 129);")
        self.tg.setObjectName("tg")
        self.cos = QtWidgets.QPushButton(self.centralwidget)
        self.cos.setGeometry(QtCore.QRect(130, 310, 101, 41))
        self.cos.setStyleSheet("background-color: rgb(109, 123, 129);")
        self.cos.setObjectName("cos")
        self.mod = QtWidgets.QPushButton(self.centralwidget)
        self.mod.setGeometry(QtCore.QRect(350, 160, 101, 41))
        self.mod.setStyleSheet("background-color: rgb(109, 123, 129);")
        self.mod.setObjectName("mod")
        self.atg = QtWidgets.QPushButton(self.centralwidget)
        self.atg.setGeometry(QtCore.QRect(20, 360, 101, 41))
        self.atg.setStyleSheet("background-color: rgb(109, 123, 129);")
        self.atg.setObjectName("atg")
        self.acos = QtWidgets.QPushButton(self.centralwidget)
        self.acos.setGeometry(QtCore.QRect(20, 260, 101, 41))
        self.acos.setStyleSheet("background-color: rgb(109, 123, 129);")
        self.acos.setObjectName("acos")
        self.asin = QtWidgets.QPushButton(self.centralwidget)
        self.asin.setGeometry(QtCore.QRect(20, 310, 101, 41))
        self.asin.setStyleSheet("background-color: rgb(109, 123, 129);")
        self.asin.setObjectName("asin")
        self.e = QtWidgets.QPushButton(self.centralwidget)
        self.e.setGeometry(QtCore.QRect(240, 160, 101, 41))
        self.e.setStyleSheet("background-color: rgb(109, 123, 129);")
        self.e.setObjectName("e")
        MainWindow.setCentralWidget(self.centralwidget)
        self.menubar = QtWidgets.QMenuBar(MainWindow)
        self.menubar.setGeometry(QtCore.QRect(0, 0, 790, 29))
        self.menubar.setObjectName("menubar")
        self.menuSmartCalc = QtWidgets.QMenu(self.menubar)
        self.menuSmartCalc.setObjectName("menuSmartCalc")
        MainWindow.setMenuBar(self.menubar)
        self.statusbar = QtWidgets.QStatusBar(MainWindow)
        self.statusbar.setObjectName("statusbar")
        MainWindow.setStatusBar(self.statusbar)
        self.menubar.addAction(self.menuSmartCalc.menuAction())

        self.retranslateUi(MainWindow)
        QtCore.QMetaObject.connectSlotsByName(MainWindow)

        self.add_functions()

    def retranslateUi(self, MainWindow):
        _translate = QtCore.QCoreApplication.translate
        MainWindow.setWindowTitle(_translate("MainWindow", "SmartCalc"))
        self.label.setText(_translate("MainWindow", "0"))
        self.zero.setText(_translate("MainWindow", "0"))
        self.dub.setText(_translate("MainWindow", "."))
        self.result.setText(_translate("MainWindow", "="))
        self.plus.setText(_translate("MainWindow", "+"))
        self.button1.setText(_translate("MainWindow", "1"))
        self.button4.setText(_translate("MainWindow", "4"))
        self.button7.setText(_translate("MainWindow", "7"))
        self.button2.setText(_translate("MainWindow", "2"))
        self.button8.setText(_translate("MainWindow", "8"))
        self.button5.setText(_translate("MainWindow", "5"))
        self.button6.setText(_translate("MainWindow", "6"))
        self.button3.setText(_translate("MainWindow", "3"))
        self.button9.setText(_translate("MainWindow", "9"))
        self.minus.setText(_translate("MainWindow", "-"))
        self.divivsion.setText(_translate("MainWindow", "/"))
        self.miltiply.setText(_translate("MainWindow", "*"))
        self.drop.setText(_translate("MainWindow", "AC"))
        self.leftBR.setText(_translate("MainWindow", "("))
        self.rightBR.setText(_translate("MainWindow", ")"))
        self.ln.setText(_translate("MainWindow", "ln"))
        self.log.setText(_translate("MainWindow", "log"))
        self.sqrt.setText(_translate("MainWindow", "sqrt"))
        self.square.setText(_translate("MainWindow", "^"))
        self.sin.setText(_translate("MainWindow", "sin"))
        self.tg.setText(_translate("MainWindow", "tg"))
        self.cos.setText(_translate("MainWindow", "cos"))
        self.mod.setText(_translate("MainWindow", "mod"))
        self.atg.setText(_translate("MainWindow", "atg"))
        self.acos.setText(_translate("MainWindow", "acos"))
        self.asin.setText(_translate("MainWindow", "asin"))
        self.e.setText(_translate("MainWindow", "e"))
        self.menuSmartCalc.setTitle(_translate("MainWindow", "SmartCalc"))

    def add_functions(self):
     self.button1.clicked.connect(lambda: self.write_number(self.button1.text()))
     self.button2.clicked.connect(lambda: self.write_number(self.button2.text()))
     self.button3.clicked.connect(lambda: self.write_number(self.button3.text()))
     self.button4.clicked.connect(lambda: self.write_number(self.button4.text()))
     self.button5.clicked.connect(lambda: self.write_number(self.button5.text()))
     self.button6.clicked.connect(lambda: self.write_number(self.button6.text()))
     self.button7.clicked.connect(lambda: self.write_number(self.button7.text()))
     self.button8.clicked.connect(lambda: self.write_number(self.button8.text()))
     self.button9.clicked.connect(lambda: self.write_number(self.button9.text()))
     self.zero.clicked.connect(lambda: self.write_number(self.zero.text()))
     self.dub.clicked.connect(lambda: self.write_number(self.dub.text()))
     self.plus.clicked.connect(lambda: self.write_number(self.plus.text()))
     self.minus.clicked.connect(lambda: self.write_number(self.minus.text()))
     self.divivsion.clicked.connect(lambda: self.write_number(self.divivsion.text()))
     self.miltiply.clicked.connect(lambda: self.write_number(self.miltiply.text()))
     self.square.clicked.connect(lambda: self.write_number(self.square.text()))
     self.leftBR.clicked.connect(lambda: self.write_number(self.leftBR.text()))
     self.rightBR.clicked.connect(lambda: self.write_number(self.rightBR.text()))
     self.ln.clicked.connect(lambda: self.write_number(self.ln.text()))
     self.log.clicked.connect(lambda: self.write_number(self.log.text()))
     self.sqrt.clicked.connect(lambda: self.write_number(self.sqrt.text()))
     self.sin.clicked.connect(lambda: self.write_number(self.sin.text()))
     self.tg.clicked.connect(lambda: self.write_number(self.tg.text()))
     self.mod.clicked.connect(lambda: self.write_number(self.mod.text()))
     self.cos.clicked.connect(lambda: self.write_number(self.cos.text()))
     self.atg.clicked.connect(lambda: self.write_number(self.atg.text()))
     self.acos.clicked.connect(lambda: self.write_number(self.acos.text()))
     self.asin.clicked.connect(lambda: self.write_number(self.asin.text()))
     self.e.clicked.connect(lambda: self.write_number(self.e.text()))
     self.drop.clicked.connect(lambda: self.label.setText(""))

    def write_number(self, number):
        # Метод по помещению строки в поле ввода
        if self.label.text() == "0":
            self.label.setText(number)
        elif number.isdigit() or number == '.':
            self.label.setText(self.label.text() + number)
        else:
            self.label.setText(self.label.text()+' '+number+' ')